### PR TITLE
Show a pop up map preview when image clicked

### LINKED ISSUE #2595: Show map preview in full growth when clicking on the map preview from any place

### PR DESCRIPTION
**Pull request by Marc-Spector**
Closes #2595 

<details>

<summary>Demo</summary>

demo3

</details>

### CODE REVIEW

**Comment by Sheikah45** [comment]
I would use the dialogLayout rather than a raw stage to keep the formatting the same

**Comment by Marc-Spector** [comment]
The dialog has a big minus - you need to specify a parent with the `StackPane` class. I used the popup instead of the dialog. It seems to be the best option.

File: src/main/java/com/faforever/client/replay/ReplayDetailController.java [hunk 1]
```diff
@@ -177,6 +177,8 @@ public void initialize() {
     qualityLabel.setTooltip(new Tooltip(i18n.get("replay.qualityTooltip")));
     deleteButton.setTooltip(new Tooltip(i18n.get("replay.deleteButton.tooltip")));
     reviewsController.setReviewSupplier(ReplayReviewBean::new);
+    mapThumbnailImageView.setOnMouseClicked(event -> Optional.ofNullable(replay.getMapVersion())
+        .ifPresent(map -> mapService.showMapInCloseUp(map.getFolderName())));
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/replay/ReplayDetailController.java]: Make this into a function and set the onMouseClicked in the fxml

File: src/main/java/com/faforever/client/map/MapService.java [hunk 2]
```diff
@@ -631,6 +637,28 @@ public void destroy() {
...
+    mapImageView.setFitHeight(imageOptimalSize);
+    mapImageView.setSmooth(true);
+    mapImageView.setPreserveRatio(true);
+
+    Popup popup = new Popup();
+    popup.setAutoHide(true);
+    popup.setAutoFix(true);
+    popup.getScene().setRoot(new StackPane(mapImageView));
+    popup.show(StageHolder.getStage(), centerScreenX, centerScreenY);
+  }
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/map/MapService.java]: I agree that this should be moved to some other util service and made general. The Services should not be creating UI elements.

Ideally the mapService would load the image and then a util class would be called that creates the popup all from the Controller Class that has the click function.

File: src/main/java/com/faforever/client/map/MapDetailController.java [hunk 3]
```diff
@@ -128,6 +128,8 @@ public void initialize() {
     };
 
     reviewsController.setReviewSupplier(MapVersionReviewBean::new);
+    thumbnailImageView.setOnMouseClicked(event -> Optional.ofNullable(mapVersion)
+        .ifPresent(map -> mapService.showMapInCloseUp(map.getFolderName())));
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/map/MapDetailController.java]: Make this a function and set the onMouseClicked in the fxml

File: src/main/java/com/faforever/client/game/GameDetailController.java [hunk 4]
```diff
@@ -62,6 +63,8 @@
 
   public void initialize() {
     watchButton = watchButtonController.getRoot();
+    mapImageView.setOnMouseClicked(event ->
+        Optional.ofNullable(game.get()).ifPresent(gameBean -> mapService.showMapInCloseUp(gameBean.getMapFolderName())));
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/game/GameDetailController.java]: Make this a function and set the onMouseClicked in the fxml

**Review by Sheikah45 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/util/PopupUtil.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import javafx.geometry.Rectangle2D;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Popup;
+import javafx.stage.Screen;
+
+public class PopupUtil {
+
+  public static void showImageInCloseUp(Image image) {
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/util/PopupUtil.java]:
```suggestion
  public static void showImagePopup(Image image) {
```

File: src/main/java/com/faforever/client/util/PopupUtil.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    showImageInCloseUp(image, 0.7);
+  }
+
+  /**
+   * @param imageHeightInPercentage Image height as a percentage of the screen height size
+   */
+  public static void showImageInCloseUp(Image image, double imageHeightInPercentage) {
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/util/PopupUtil.java]:
```suggestion
  public static void showImagePopup(Image image, double imageHeightInPercentage) {
```

**Review by Sheikah45 — CHANGES_REQUESTED** [review]

**Review by Sheikah45 — APPROVED** [review]

**Pull request merged**